pkg/envoy/proxy.go

```diff
@@ -90,14 +94,24 @@ func (p *Proxy) SetNewNonce(typeURI TypeURI) string {
 
 // String returns the CommonName of the proxy.
 func (p Proxy) String() string {
-	return string(p.GetCommonName())
+	return string(p.GetCertificateCommonName())
+}
+
+// GetPodUID returns the UID of the pod, which the connected Envoy proxy is fronting.
+func (p Proxy) GetPodUID() string {
+	return p.PodUID
 }
 
-// GetCommonName returns the Subject Common Name from the mTLS certificate of the Envoy proxy connected to xDS.
-func (p Proxy) GetCommonName() certificate.CommonName {
+// GetCertificateCommonName returns the Subject Common Name from the mTLS certificate of the Envoy proxy connected to xDS.
+func (p Proxy) GetCertificateCommonName() certificate.CommonName {
 	return p.CommonName
 }
 
+// GetCertificateSerialNumber returns the Serial Number of the certificate for the connected Envoy proxy.
```

Comment:
This method on a `Proxy` object can be confusing because of the different types of certs associated with a proxy. My concern here is not regarding the function name, but more to do with exposing a serial no. api on a proxy object, which is to do with internals of a certificate.

Previously though, one could argue the CN being an identifier for the proxy. Can we abstract this to `GetProxyUID()` and internally leverage the pod UID/XDS cert serial no. instead?

Reply:
@shashankram I added a `GetPodUID()` in this PR as well. The goal is going to be to use both - the SerialNumber of the xDS certificate and also log the UID of the Pod, which the proxy is fronting.

On the one hand logging SerialNumber and PodUID seems like a lot - but I know that when time comes to troubleshoot it we will be thankful this extra context was available.